pkg/controller/jobs/jobset/jobset_webhook.go

```diff
@@ -80,9 +82,25 @@ func (w *JobSetWebhook) Default(ctx context.Context, obj runtime.Object) error {
 
 	jobframework.ApplyDefaultForManagedBy(jobSet, w.queues, w.cache, log)
 
+	applyDefaultSliceSize(jobSet)
+
 	return nil
 }
 
+func applyDefaultSliceSize(jobSet *JobSet) {
+	for _, rj := range jobSet.Spec.ReplicatedJobs {
+		targetSliceSize := rj.Template.Spec.Parallelism
+		annotations := rj.Template.Spec.Template.Annotations
+
+		_, sliceSizeFound := annotations[kueuealpha.PodSetSliceSizeAnnotation]
+		_, sliceRequiredFound := annotations[kueuealpha.PodSetSliceRequiredTopologyAnnotation]
+
+		if sliceRequiredFound && !sliceSizeFound && targetSliceSize != nil {
+			annotations[kueuealpha.PodSetSliceSizeAnnotation] = strconv.Itoa(int(*targetSliceSize))
```

Comment:
This will panic if not parsable and probably crash entire Kueue, so it could be use by malicious actors. Please parse it and return error to the user via `error`. For parsing use same helpers similar to, but for annotations: https://github.com/kubernetes-sigs/kueue/blob/d2f9f2efcd066f007930e9f04fe7629ed8983a0a/pkg/util/pod/pod.go#L81-L103